static/usr/lib/systemd/systemd-shutdown.wrapper

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# We need to manually move /run/mnt/data after shutdown switch root
+# so that systemd-shutdown can umount /oldroot in the shutdown initramfs.
+# systemd-shutdown has hooks, but they are run either before
+# switch, or after unmounts.
+# It would be better to make systemd-shutdown smarter and move
+# all found mounts before trying to unmount them.
```

Comment:
somehow i would rather try to fix this in systemd-shutdown itself, instead of adding all of this into core20. Given that systemd-shutdown is C and can execute all of these things.

or for example, if some else can be done as a finalrd hook easier.

For example, can we have finalrd/core.finalrd hook that simply does `umount --lazy /run/mnt/*` ?

Reply:
Yes, I will try to change `systemd-shutdown` to move filesystems in sub mount points. It should be an acceptable improvement for upstream.

The issue with `--lazy` is that it does not give you an error if it does not manage to unmount before shutdown. At least I am not aware of any way to check the active detached mounts. If only we had that, we could just mark `run-mnt-base.mount` as `LazyUmount=yes`, and verify it was really unmounted in a `system-shutdown` hook after `systemd-shutdown` has umounted `/sysroot`.